Classes/RealUrl/TitleMapping.php

```diff
@@ -0,0 +1,233 @@
+     * @return Utility
+     * @throws \ReflectionException
+     */
+    protected function getUtilityFromDecoder(): Utility
+    {
+        $urlDecoderReflection = new \ReflectionClass($this->urlDecoder);
+        $reflectionProperty = $urlDecoderReflection->getProperty('utility');
+        $reflectionProperty->setAccessible(true);
+        return $reflectionProperty->getValue($this->urlDecoder);
+    }
+
+    /**
+     * Get repository for requested controller
+     *
+     * @param string $controller
+     * @return AbstractRepository
+     * @throws \Exception if controller name is invalid
+     */
+    protected function getRepositoryForController(string $controller): AbstractRepository
+    {
+        switch($controller) {
+            case 'lebenslagen':
+                $repository = $this->objectManager->get(LebenslagenRepository::class);
+                break;
+            case 'leistungen':
+                $repository = $this->objectManager->get(LeistungenRepository::class);
+                break;
+            case 'organisationseinheiten':
+                $repository = $this->objectManager->get(OrganisationseinheitenRepository::class);
+                break;
+            default:
+                throw new \InvalidArgumentException(
+                    'Could not find repository for selected controller "' . $controller . '"!',
+                    1523960421
+                );
+                break;
+        }
+        return $repository;
+    }
+
+    /**
+     * Returns the title as url friendly string or $id if no title could be found
+     *
+     * @param int $id
+     * @param string $controller
+     * @return string
+     */
+    protected function encodeIdToTitle(int $id, string $controller): string
+    {
+        $repository = $this->getRepositoryForController($controller);
+        switch($controller) {
+            case 'lebenslagen':
+                $title = $repository->getById($id)['displayName'];
+                break;
+            case 'leistungen':
+                $title = $repository->getLiveById($id)['title'];
+                break;
+            case 'organisationseinheiten':
+                $title = $repository->getById($id)['name'];
+                break;
+            default:
+                $title = '';
+                break;
+        }
+        // make it url friendly
+        return $this->utility->convertToSafeString($title ?: (string)$id);
+    }
+
+    /**
+     * Decode title to id
+     *
+     * @param string $title
+     * @return int id or -1 if no id was found!
+     */
+    protected function decodeTitleToId(string $title): int
+    {
+        $controller = $this->getControllerAliasFromUrlDecoder();
+        $repository = $this->getRepositoryForController($controller);
+        switch ($controller) {
+            case 'lebenslagen':
+                /** @var LebenslagenRepository $repository */
+                $id = $this->findIdRecursively($title, 'name', $repository->getAll());
```

Comment:
Wow...What are you doing? Collecting 12.400 records? Maybe try to modify query afterwards or call me...maybe I have a better idea